src/test/stl_loader_test.cpp

```diff
@@ -1,44 +1,51 @@
 #include <gtest/gtest.h>
-#include <boost/filesystem.hpp>
+#include <rviz/mesh_loader.h>
 #include <ros/package.h>
-#include <rviz/ogre_helpers/stl_loader.h>
+#include <rviz/ogre_helpers/render_system.h>
 
 TEST( STLLoader, load )
 {
   // Get the path to the directory where the meshes are located.
-  boost::filesystem::path meshDir = ros::package::getPath("rviz");
-  meshDir /= "src/test/meshes";
+  std::string meshDir = "package://rviz/src/test/meshes/";
 
-  // Load an ascii STL file. Note that only binary STL files are supported.
-  ogre_tools::STLLoader loader;
-  boost::filesystem::path meshFilePath = meshDir / "ascii.stl";
-  EXPECT_FALSE(loader.load(meshFilePath.string()));
+  // Load a valid ascii STL file.
+  std::string meshFilePath = meshDir + "valid_ascii.stl";
+  EXPECT_TRUE(rviz::loadMeshFromResource(meshFilePath).get());
+
+  // Load an ascii STL file. Note that this file uses "end facet" instead of "endfacet"
+  // and therefore fails in assimp
+  meshFilePath = meshDir + "ascii.stl";
+  EXPECT_FALSE(rviz::loadMeshFromResource(meshFilePath).get());
 
   // Load an invalid STL binary file (size < 84 bytes).
-  meshFilePath = meshDir / "invalid_short.stl";
-  EXPECT_FALSE(loader.load(meshFilePath.string()));
+  meshFilePath = meshDir + "invalid_short.stl";
+  EXPECT_FALSE(rviz::loadMeshFromResource(meshFilePath).get());
 
   // Load an invalid STL binary file (size does not match the expected size).
-  meshFilePath = meshDir / "invalid.stl";
-  EXPECT_FALSE(loader.load(meshFilePath.string()));
+  meshFilePath = meshDir + "invalid.stl";
+  EXPECT_FALSE(rviz::loadMeshFromResource(meshFilePath).get());
 
   // Load an invalid STL binary file (size does not match the expected size,
   // but does if incorrectly read as an 16-bit uint)
-  meshFilePath = meshDir / "16bit_vs_32bit_should_fail.stl";
-  EXPECT_FALSE(loader.load(meshFilePath.string()));
+  meshFilePath = meshDir + "16bit_vs_32bit_should_fail.stl";
+  EXPECT_FALSE(rviz::loadMeshFromResource(meshFilePath).get());
 
   // Load a valid STL binary file.
-  meshFilePath = meshDir / "valid.stl";
-  EXPECT_TRUE(loader.load(meshFilePath.string()));
+  meshFilePath = meshDir + "valid.stl";
+  EXPECT_TRUE(rviz::loadMeshFromResource(meshFilePath).get());
 
   // Load a "potentially" valid STL binary file with bigger size than the
   // expected. The extra "unexpected" data at the end of the file should be
-  // ignored.
-  meshFilePath = meshDir / "valid_extra.stl";
-  EXPECT_TRUE(loader.load(meshFilePath.string()));
+  // ignored. Maybe. But is not ignored by assimp
```

Comment:
I'm a little bit hesitant to merge this as is, because of this "regression".
Formally, of course, proper .stl files shouldn't have excess data at the end of the file.
However, I'm not sure how many users are out there that have such broken .stl files and rely on rviz to load them anyway. @wjwwood What is your recommendation?

Reply:
Well at least MoveIt ( https://github.com/ros-planning/geometric_shapes/blob/melodic-devel/src/mesh_operations.cpp ) uses assimp directly as well so these files would not work in a MoveIt context